src/coverlet.collector/InProcDataCollection/CoverletInProcDataCollector.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Reflection;
+using Coverlet.Core.Instrumentation;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.DataCollection;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.DataCollector.InProcDataCollector;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.InProcDataCollector;
+
+namespace Coverlet.Collector.InProcDataCollector
+{
+    public class CoverletInProcDataCollector : InProcDataCollection
+    {
+        public void Initialize(IDataCollectionSink dataCollectionSink)
+        {
+        }
+
+        public void TestCaseEnd(TestCaseEndArgs testCaseEndArgs)
+        {
+        }
+
+        public void TestCaseStart(TestCaseStartArgs testCaseStartArgs)
+        {
+        }
+
+        public void TestSessionEnd(TestSessionEndArgs testSessionEndArgs)
+        {
+            foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+            {
+                Type injectedInstrumentationClass = GetInstrumentationClass(assembly);
+                if (injectedInstrumentationClass is null)
+                    continue;
+
+                var unloadModule = injectedInstrumentationClass.GetMethod(nameof(ModuleTrackerTemplate.UnloadModule), new[] { typeof(object), typeof(EventArgs) });
```

Comment:
Is this possible?And if we found instrumentation class but now `UnloadModule` we should `continue` or maybe throw or log? cc: @tonerdo

Reply:
@tonerdo Should we throw? Adding logs for better traceability.

Reply:
Added logs for debugging. Not sure if we should need to throw.

Reply:
IMO we should trow(raise error) if we don't expect correct state
* throw if we don't found `UnloadModule`(I don't think should be possible), and if in future we change something thanks to nameof compiler will help.
* throw if we get exception from `UnloadModule`, here we save hits, it must work or result could be not correct(corruption at "correct" place)

Reply:
```
if (unloadModule is null)
   continue;
```
If I am getting this right, the above shouldn't occur. If it is unable to find the unload module or any exception in unload module [here](https://github.com/tonerdo/coverlet/blob/master/src/coverlet.core/Instrumentation/ModuleTrackerTemplate.cs#L124), should be thrown, which makes sense.

Reply:
@MarcoRossignoli PTAL. Throwing the exception now.

Reply:
I like it!